### PR TITLE
Fixes #32134 - client certificate support for smart proxy

### PR DESCRIPTION
**Pull request by lzap**
This allows running webhooks over HTTPS against smart proxies running shellhooks.

### CODE REVIEW

**Review by ofedoren — CHANGES_REQUESTED** [review]
Thanks, @lzap! LGTM, but I have just one thing to ask, see inline.

File: app/controllers/concerns/foreman_webhooks/controller/parameters/webhook.rb [hunk 1]
```diff
@@ -20,7 +20,8 @@ def webhook_params_filter
                             :ssl_ca_certs,
                             :user,
                             :password,
-                            :http_headers
+                            :http_headers,
+                            :proxy_authorization
```
[the remark this answers is not in the document]
Comment (lzap) [reply app/controllers/concerns/foreman_webhooks/controller/parameters/webhook.rb:24]: Done.

**Review by ofedoren — COMMENTED** [review]
Thanks, @lzap. LGTM and seems to be working, although just one nipick, please see inline.

File: app/controllers/api/v2/webhooks_controller.rb [hunk 2]
```diff
@@ -34,6 +34,7 @@ def show; end
           param :user, String
           param :password, String
           param :http_headers, String
+          param :proxy_authorization, String, N_('Authorize with Foreman client certificate and validate smart-proxy CA from Settings')
```
[the remark this answers is not in the document]
Comment (lzap) [reply app/controllers/api/v2/webhooks_controller.rb]: Fixed good catch.

**Review by ofedoren — APPROVED** [review]
Thanks, @lzap! Merging.

**Comment by lzap** [comment]
Thanks right on time!

**Pull request merged**